pkg/crc/validation/validation.go

```diff
@@ -98,6 +105,30 @@ func ValidateIPAddress(ipAddress string) error {
 	return nil
 }
 
+func ValidateURL(inputString string) bool {
+	u, err := url.ParseRequestURI(inputString)
+	if err != nil {
+		return false
+	}
+	if u.Scheme != "http" && u.Scheme != "https" && u.Scheme != "docker" {
+		return false
+	}
+	return true
+}
+
+func validRegistryURI(inputString string) error {
```

Comment:
Why is it not part of `validateURL`?

Reply:
@cfergeau As of now I am using `ValidateURL` only for scheme validation and if not vaildURI then I check for validpath.

Reply:
Yes, but why do it this way? Isn't it more convenient to do it all in one go in `ValidateURL`? Is there a drawback to doing this?

Reply:
@cfergeau No but then I have to check the validpath also as part of validURL function something like so need to change the function name something like `ValidateUrlOrPath` ?
```golang
func ValidateURL(inputString string) error {
	u, err := url.ParseRequestURI(inputString)
	if err != nil {
		return ValidatePath(inputString)
	}
	switch {
	case u.Scheme == "http", u.Scheme == "https":
		return nil
	case u.Scheme == "docker":
		if len(strings.Split(inputString, ":")) != 3 {
			return fmt.Errorf("invalid %s registry URL, tag is required (such as docker://quay.io/crcont/ocp-bundle:4.11.0)", inputString)
		}
		return nil
	default:
		return fmt.Errorf("invalid %s format (only supported http, https or docker)", inputString)
	}
	return nil
}
```

Reply:
You don't necessarily have to put `ValidatePath` in it, but I'd prefer to avoid having something which looks like `if ValidateURL() { if reallyValidateURL() {}}`, better to do all the URL validation at once.